requirements.txt

```diff
@@ -6,3 +6,5 @@ watchdog==0.8.3
 # Use sckoo@stanford.edu's patched argcomplete, until pull request has been accepted
 # https://github.com/kislyuk/argcomplete/pull/118
 git+https://github.com/kashizui/argcomplete.git@allow-overridden-subparsersaction
+# Use agd@stanford.edu's fork of hash_ring with extra features
+git+https://github.com/andreweduffy/hash_ring.git@master
```

Comment:
Actually the setup.py file apparently states that the project is under the BSD license (though no explicit license is included with the source), so that means that we can reproduce and use it. I can definitely make this a CodaLab-official maintained fork if you think that'd be best. I can also go ahead and write one from scratch based off of another language's implementation, let me know what you think.